ua-frontend/libs/ua-core/src/+state/ui.reducer.ts

```diff
@@ -0,0 +1,13 @@
+import { Ui } from './ui.interfaces';
+import { UiAction } from './ui.actions';
+
+export function uiReducer(state: Ui, action: UiAction): Ui {
+  switch (action.type) {
+    case 'DATA_LOADED': {
+      return { ...state, ...action.payload };
```

Comment:
When possible we should be explicit about what we set from the payload onto the state. The problem comes when the payload unexpectedly has something that messes things up downstream.